exporters/memory/src/in_memory_span_exporter.cc

```diff
@@ -0,0 +1,43 @@
+#include "opentelemetry/exporters/memory/in_memory_span_exporter.h"
+
+#include <iostream>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+InMemorySpanExporter::InMemorySpanExporter(
+    std::shared_ptr<std::vector<std::unique_ptr<sdk::trace::SpanData>>> spans_received,
+    std::shared_ptr<bool> shutdown_called)
+    : spans_received_(spans_received), shutdown_called_(shutdown_called)
+{}
+
+std::unique_ptr<sdk::trace::Recordable> InMemorySpanExporter::MakeRecordable() noexcept
+{
+  return std::unique_ptr<sdk::trace::Recordable>(new sdk::trace::SpanData);
+}
+
+sdk::trace::ExportResult InMemorySpanExporter::Export(
+    const nostd::span<std::unique_ptr<sdk::trace::Recordable>> &recordables) noexcept
+{
+  for (auto &recordable : recordables)
+  {
+    auto span = std::unique_ptr<sdk::trace::SpanData>(
+        static_cast<sdk::trace::SpanData *>(recordable.release()));
```

Comment:
I left the same comment in Oliver's documentation PR, so copying here:

Hm, this takes ownership of the recordable object. This is fairly dangerous, since the param "recordables" will no longer have ownership of its Recordable members. Note that unique_ptr.release() is not a const method. I believe the compiler allows this because you're not mutating the input param "recordables", just the container objects (this is just my theory though).

The implementation right now makes Export take ownership over all recordables passed to it. Consider if this is the desired behavior for this exporter.

Reply:
Addressed offline and was told to disregard this comment.